<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 3 Image segmentation | Visium Data Preprocessing</title>
  <meta name="description" content="Online book ‘Visium Data Preprocessing’" />
  <meta name="generator" content="bookdown 0.35 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 3 Image segmentation | Visium Data Preprocessing" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="Online book ‘Visium Data Preprocessing’" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 3 Image segmentation | Visium Data Preprocessing" />
  
  <meta name="twitter:description" content="Online book ‘Visium Data Preprocessing’" />
  



<meta name="date" content="2023-09-21" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="visium-data-preprocessing.html"/>
<link rel="next" href="loupe-browser.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Visium Data Preprocessing</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Welcome</a></li>
<li class="part"><span><b>I Introduction</b></span></li>
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="introduction.html"><a href="introduction.html#spatial-transcriptomics"><i class="fa fa-check"></i><b>1.1</b> Spatial transcriptomics</a></li>
<li class="chapter" data-level="1.2" data-path="introduction.html"><a href="introduction.html#x-genomics-visium"><i class="fa fa-check"></i><b>1.2</b> 10x Genomics Visium</a></li>
</ul></li>
<li class="part"><span><b>II Visium data preprocessing</b></span></li>
<li class="chapter" data-level="2" data-path="visium-data-preprocessing.html"><a href="visium-data-preprocessing.html"><i class="fa fa-check"></i><b>2</b> Visium data preprocessing</a></li>
<li class="chapter" data-level="3" data-path="image-segmentation.html"><a href="image-segmentation.html"><i class="fa fa-check"></i><b>3</b> Image segmentation</a>
<ul>
<li class="chapter" data-level="3.1" data-path="image-segmentation.html"><a href="image-segmentation.html#overview"><i class="fa fa-check"></i><b>3.1</b> Overview</a></li>
<li class="chapter" data-level="3.2" data-path="image-segmentation.html"><a href="image-segmentation.html#what-is-vistoseg"><i class="fa fa-check"></i><b>3.2</b> What is VistoSeg?</a></li>
<li class="chapter" data-level="3.3" data-path="image-segmentation.html"><a href="image-segmentation.html#split-histology-images"><i class="fa fa-check"></i><b>3.3</b> Split histology images</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="image-segmentation.html"><a href="image-segmentation.html#file-size-issues"><i class="fa fa-check"></i><b>3.3.1</b> File size issues</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="image-segmentation.html"><a href="image-segmentation.html#segment-cell-nuclei"><i class="fa fa-check"></i><b>3.4</b> Segment cell nuclei</a></li>
<li class="chapter" data-level="3.5" data-path="image-segmentation.html"><a href="image-segmentation.html#identify-number-of-cells-per-spot"><i class="fa fa-check"></i><b>3.5</b> Identify number of cells per spot</a></li>
<li class="chapter" data-level="3.6" data-path="image-segmentation.html"><a href="image-segmentation.html#wrapping-up"><i class="fa fa-check"></i><b>3.6</b> Wrapping up</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="loupe-browser.html"><a href="loupe-browser.html"><i class="fa fa-check"></i><b>4</b> Loupe Browser</a>
<ul>
<li class="chapter" data-level="4.1" data-path="loupe-browser.html"><a href="loupe-browser.html#overview-1"><i class="fa fa-check"></i><b>4.1</b> Overview</a></li>
<li class="chapter" data-level="4.2" data-path="loupe-browser.html"><a href="loupe-browser.html#manual-alignment-of-images"><i class="fa fa-check"></i><b>4.2</b> Manual alignment of images</a></li>
<li class="chapter" data-level="4.3" data-path="loupe-browser.html"><a href="loupe-browser.html#output-files-and-space-ranger"><i class="fa fa-check"></i><b>4.3</b> Output files and Space Ranger</a></li>
<li class="chapter" data-level="4.4" data-path="loupe-browser.html"><a href="loupe-browser.html#downstream-analyses-in-loupe-browser"><i class="fa fa-check"></i><b>4.4</b> Downstream analyses in Loupe Browser</a></li>
<li class="chapter" data-level="4.5" data-path="loupe-browser.html"><a href="loupe-browser.html#wrapping-up-1"><i class="fa fa-check"></i><b>4.5</b> Wrapping up</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="space-ranger.html"><a href="space-ranger.html"><i class="fa fa-check"></i><b>5</b> Space Ranger</a>
<ul>
<li class="chapter" data-level="5.1" data-path="space-ranger.html"><a href="space-ranger.html#overview-2"><i class="fa fa-check"></i><b>5.1</b> Overview</a></li>
<li class="chapter" data-level="5.2" data-path="space-ranger.html"><a href="space-ranger.html#what-is-space-ranger"><i class="fa fa-check"></i><b>5.2</b> What is Space Ranger?</a></li>
<li class="chapter" data-level="5.3" data-path="space-ranger.html"><a href="space-ranger.html#installing-space-ranger"><i class="fa fa-check"></i><b>5.3</b> Installing Space Ranger</a></li>
<li class="chapter" data-level="5.4" data-path="space-ranger.html"><a href="space-ranger.html#run-spaceranger-count"><i class="fa fa-check"></i><b>5.4</b> Run spaceranger count</a></li>
<li class="chapter" data-level="5.5" data-path="space-ranger.html"><a href="space-ranger.html#output-files"><i class="fa fa-check"></i><b>5.5</b> Output files</a></li>
<li class="chapter" data-level="5.6" data-path="space-ranger.html"><a href="space-ranger.html#web-summary-.html-file"><i class="fa fa-check"></i><b>5.6</b> Web summary .html file</a></li>
<li class="chapter" data-level="5.7" data-path="space-ranger.html"><a href="space-ranger.html#import-outputs-into-r"><i class="fa fa-check"></i><b>5.7</b> Import outputs into R</a></li>
<li class="chapter" data-level="5.8" data-path="space-ranger.html"><a href="space-ranger.html#vistoseg-for-quantifying-cells-per-spot"><i class="fa fa-check"></i><b>5.8</b> VistoSeg for quantifying cells per spot</a></li>
<li class="chapter" data-level="5.9" data-path="space-ranger.html"><a href="space-ranger.html#wrapping-up-2"><i class="fa fa-check"></i><b>5.9</b> Wrapping up</a></li>
</ul></li>
<li class="part"><span><b>III Appendix</b></span></li>
<li class="chapter" data-level="6" data-path="appendix.html"><a href="appendix.html"><i class="fa fa-check"></i><b>6</b> Appendix</a>
<ul>
<li class="chapter" data-level="6.1" data-path="appendix.html"><a href="appendix.html#contributors"><i class="fa fa-check"></i><b>6.1</b> Contributors</a></li>
<li class="chapter" data-level="6.2" data-path="appendix.html"><a href="appendix.html#github-actions-workflow"><i class="fa fa-check"></i><b>6.2</b> GitHub Actions workflow</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Visium Data Preprocessing</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="image-segmentation" class="section level1 hasAnchor" number="3">
<h1><span class="header-section-number">Chapter 3</span> Image segmentation<a href="image-segmentation.html#image-segmentation" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="overview" class="section level2 hasAnchor" number="3.1">
<h2><span class="header-section-number">3.1</span> Overview<a href="image-segmentation.html#overview" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The Visium platform from 10x Genomics has four tissue capture areas.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-3"></span>
<img src="images/Visium_Picture1.png" alt="Visium slide design. Source: [10x Genomics](https://kb.10xgenomics.com/hc/en-us/articles/360035487572-What-is-the-spatial-resolution-and-configuration-of-the-capture-area-of-the-Visium-Gene-Expression-Slide-)." width="60%" />
<p class="caption">
Figure 3.1: Visium slide design. Source: <a href="https://kb.10xgenomics.com/hc/en-us/articles/360035487572-What-is-the-spatial-resolution-and-configuration-of-the-capture-area-of-the-Visium-Gene-Expression-Slide-">10x Genomics</a>.
</p>
</div>
<p>While 10x Genomics will provide the Visium slide, you can use many different imaging scopes to obtain bright field and/or fluorescent images for regular Visium or Visium immunofluorescence (IF), respectively.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-4"></span>
<img src="images/tissue_lowres_image_brightfield.png" alt="Visium images you might generate using your preferred imaging scope: bright field images showing the tissue or multi-channel fluorescent images. Source: [10x Genomics](https://support.10xgenomics.com/spatial-gene-expression/software/pipelines/latest/using/input-recommendations)." width="75%" />
<p class="caption">
Figure 3.2: Visium images you might generate using your preferred imaging scope: bright field images showing the tissue or multi-channel fluorescent images. Source: <a href="https://support.10xgenomics.com/spatial-gene-expression/software/pipelines/latest/using/input-recommendations">10x Genomics</a>.
</p>
</div>
<p>As we will learn in the next chapters, we will want to use <a href="loupe-browser.html#loupe-browser">Loupe Browser</a> to manually align each tissue capture area prior to running <a href="space-ranger.html#space-ranger">Space Ranger</a> for processing the gene expression data.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-5"></span>
<img src="images/aligned_fiducials.jpg" alt="A single Visium tissue capture area with the fiducial frame aligned through the Loupe Browser such that the red circles overlap the fiducial frame as determined by visual inspection. Source: [10x Genomics](https://support.10xgenomics.com/spatial-gene-expression/software/pipelines/latest/output/images)." width="75%" />
<p class="caption">
Figure 3.3: A single Visium tissue capture area with the fiducial frame aligned through the Loupe Browser such that the red circles overlap the fiducial frame as determined by visual inspection. Source: <a href="https://support.10xgenomics.com/spatial-gene-expression/software/pipelines/latest/output/images">10x Genomics</a>.
</p>
</div>
<p>You likely will image a full Visium slide with four tissue capture areas. So before you can proceed, you will need to use some external software to split your image into individual capture areas. One such utility is <a href="http://research.libd.org/VistoSeg/"><code>VistoSeg</code></a>, which we will illustrate in this chapter.</p>
</div>
<div id="what-is-vistoseg" class="section level2 hasAnchor" number="3.2">
<h2><span class="header-section-number">3.2</span> What is VistoSeg?<a href="image-segmentation.html#what-is-vistoseg" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><a href="http://research.libd.org/VistoSeg/"><code>VistoSeg</code></a> is a MATLAB pipeline used to process, analyze, and interactively visualize the high-resolution histology images from the 10x Genomics Visium experiment (<a href="https://doi.org/10.1101/2021.08.04.452489">Tippani et al. (2022)</a>). One of the outputs from <code>VistoSeg</code> is the estimated number of cells per spot, which can be used as an input in downstream analyses. <code>VistoSeg</code> can also be used prior to running <code>Loupe</code> as it can help create high resolution images for each Visium capture area. The main steps in <code>VistoSeg</code> are as follows:</p>
<ul>
<li><em>Split Image</em>: This step is used to split the large Visium histology whole slide image into individual capture areas, creating four individual <code>.tif</code> files. These images can then be used with <a href="loupe-browser.html#loupe-browser">Loupe Browser</a>.</li>
<li><em>Nuclei Segmentation</em>: This is a two-step process where nuclei segmentation is performed for images. Nuclei segmentation is followed by a refining step to get a cleaner image of the nuclei segmentation.</li>
<li><em>Space Ranger</em>: Next, you need to run <code>spaceranger count</code>, described in the <a href="space-ranger.html#space-ranger">Space Ranger</a> chapter, to create the input files for the final <code>VistoSeg</code> step.</li>
<li><em>Count Cells/Nuclei</em>: This step is performed to get the count of cells/nuclei per Visium spot for each capture area.</li>
</ul>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-6"></span>
<img src="images/VistoSeg.png" alt="`VistoSeg` pipeline overview. Source: [`VistoSeg`](http://research.libd.org/VistoSeg/index.html#why-spatial-transcriptomics-or-visium-imaging)." width="100%" />
<p class="caption">
Figure 3.4: <code>VistoSeg</code> pipeline overview. Source: <a href="http://research.libd.org/VistoSeg/index.html#why-spatial-transcriptomics-or-visium-imaging"><code>VistoSeg</code></a>.
</p>
</div>
<p>For more information about <code>VistoSeg</code> please see its <a href="http://research.libd.org/VistoSeg/">documentation website</a>.</p>
</div>
<div id="split-histology-images" class="section level2 hasAnchor" number="3.3">
<h2><span class="header-section-number">3.3</span> Split histology images<a href="image-segmentation.html#split-histology-images" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><a href="http://research.libd.org/VistoSeg/"><code>VistoSeg</code></a> provides a MATLAB function called <code>splitSlide</code> that can take an image with four capture areas and divide this into four images with approximately the same dimensions (they can be a few pixels off in the x-axis). Sometimes the center of the image is not the center of the slide, in which case the offset can be adjusted manually.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-7"></span>
<img src="images/VistoSeg_img1.png" alt="An example bright field image with four Visium tissue capture areas. Source: [`VistoSeg`](http://research.libd.org/VistoSeg/step-1-split-visium-histology-whole-slide-image-into-individual-capture-area-images.html)." width="100%" />
<p class="caption">
Figure 3.5: An example bright field image with four Visium tissue capture areas. Source: <a href="http://research.libd.org/VistoSeg/step-1-split-visium-histology-whole-slide-image-into-individual-capture-area-images.html"><code>VistoSeg</code></a>.
</p>
</div>
<p>Example images for trying out <code>splitSplide</code> and the corresponding code are available from the <a href="http://research.libd.org/VistoSeg/step-1-split-visium-histology-whole-slide-image-into-individual-capture-area-images.html"><code>VistoSeg</code> documentation website</a>.</p>
<div id="file-size-issues" class="section level3 hasAnchor" number="3.3.1">
<h3><span class="header-section-number">3.3.1</span> File size issues<a href="image-segmentation.html#file-size-issues" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>As noted on the <a href="http://research.libd.org/VistoSeg/index.html#software-requirements"><code>VistoSeg</code> software requirements</a> page, high-quality images for a full Visium slide can be as large as ~25 GB each. This means that you will need plenty of memory in order to load the full Visium slide image into MATLAB before you can run <code>splitSlide</code>. This could be two or three times the amount of the image size on disk. Once you have run <code>splitSlide</code>, the rest of the <code>VistoSeg</code> pipeline can run on a machine with 16 GB of RAM (or even less).</p>
</div>
</div>
<div id="segment-cell-nuclei" class="section level2 hasAnchor" number="3.4">
<h2><span class="header-section-number">3.4</span> Segment cell nuclei<a href="image-segmentation.html#segment-cell-nuclei" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Once you have run the <code>splitSlide</code> function from <code>VistoSeg</code>, you can now proceed to segment the images. This process helps remove background noise in the images so you can more readily identify the cells or nuclei in your bright field image. For fluorescent images, <code>dotdotdot</code>’s utilities (<a href="https://doi.org/10.1093/nar/gkaa312">Maynard et al. (2020)</a>) could potentially be useful, particularly if your images have Z-stacks.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-8"></span>
<img src="images/dotdotdot_fig1.jpeg" alt="`dotdotdot` overview figure. Notice the Z-stacked images which some scopes produce. Source: [`dotdotdot`](https://doi.org/10.1093/nar/gkaa312)." width="80%" />
<p class="caption">
Figure 3.6: <code>dotdotdot</code> overview figure. Notice the Z-stacked images which some scopes produce. Source: <a href="https://doi.org/10.1093/nar/gkaa312"><code>dotdotdot</code></a>.
</p>
</div>
<p><code>VistoSeg</code> provides two functions: <code>VNS</code> (Visium Nuclei Segmentation) and <code>refineVNS</code> which are used to perform the cell/nuclei segmentation on regular bright field images. This process involves choosing the total number of visually distinguishable colors in the image (5 in our sample image) and running MATLAB’s <a href="https://www.mathworks.com/help/images/color-based-segmentation-using-k-means-clustering.html">color-based segmentation using k-means clustering</a>.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-9"></span>
<img src="images/VistoSeg_sampleimage_index3.png" alt="Color cluster 3 is the one that best corresponds to the nuclei for an example capture area image shown in the `VistoSeg` documentation. Pixels corresponding to this cluster are shown in gray and the remaining background pixels in green. Source: [`VistoSeg`](http://research.libd.org/VistoSeg/step-1-split-visium-histology-whole-slide-image-into-individual-capture-area-images.html)." width="75%" />
<p class="caption">
Figure 3.7: Color cluster 3 is the one that best corresponds to the nuclei for an example capture area image shown in the <code>VistoSeg</code> documentation. Pixels corresponding to this cluster are shown in gray and the remaining background pixels in green. Source: <a href="http://research.libd.org/VistoSeg/step-1-split-visium-histology-whole-slide-image-into-individual-capture-area-images.html"><code>VistoSeg</code></a>.
</p>
</div>
<p>For more information and a detailed example check the <a href="http://research.libd.org/VistoSeg/step-1-split-visium-histology-whole-slide-image-into-individual-capture-area-images.html"><code>VistoSeg</code> documentation website</a>.</p>
</div>
<div id="identify-number-of-cells-per-spot" class="section level2 hasAnchor" number="3.5">
<h2><span class="header-section-number">3.5</span> Identify number of cells per spot<a href="image-segmentation.html#identify-number-of-cells-per-spot" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><code>VistoSeg</code> has two utilities for estimating the number of cells per spot. As noted above, for this step you first need to also run <a href="loupe-browser.html#loupe-browser">Loupe Browser</a> and <a href="space-ranger.html#space-ranger">Space Ranger</a> to generate the Space Ranger output files, which are required here as an input.</p>
<p>The alignment of the individual tissue capture areas with <code>Loupe Browser</code> is required to identify the centers of each of the Visium spots and their coordinates in the images. As long as all downstream segmented images use the exact same proportions as the input image for <code>Loupe Browser</code>, we can compute a scaling factor and use it for computing the coordinates of the Visium spots in the new images. <code>Space Ranger</code> will then provide us the spot coordinate file as well as process the gene expression data. The spot coordinate files will enable us to estimate the number of cells per spot with <code>VistoSeg</code>, as shown in the <a href="space-ranger.html#space-ranger">Space Ranger</a> chapter.</p>
</div>
<div id="wrapping-up" class="section level2 hasAnchor" number="3.6">
<h2><span class="header-section-number">3.6</span> Wrapping up<a href="image-segmentation.html#wrapping-up" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>In this chapter we learned about image processing steps required to work with the images we generate as part of our Visium experiments. Given the broad spectrum of imaging scopes, these tools are not available directly within the software provided by 10x Genomics. <a href="http://research.libd.org/VistoSeg/"><code>VistoSeg</code></a> can help us split our images and process them so that we can estimate the number of cells or nuclei per Visium spot, which we can then use as an input during downstream analyses.</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="visium-data-preprocessing.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="loupe-browser.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
},
"lib_dir": "book_assets"
});
});
</script>

</body>

</html>
