# (PART) Introduction {-}

# Introduction

This book describes a computational analysis pipeline for spatially resolved transcriptomics (ST) data, using the [Bioconductor](http://bioconductor.org/) framework and R programming language.


## Contents of this book

The chapters are organized into several parts:

- **Introduction:** introduction, spatially resolved transcriptomics, and the `SpatialExperiment` object class

- **Preprocessing steps:** preprocessing steps to prepare raw data for loading into R

- **Analysis steps:** chapters describing key analysis steps using the Bioconductor framework, including examples of R code that can be run on your own laptop

- **Workflows:** complete workflows for several example datasets

- **Appendix:** contributors, acknowledgments, references


## Who this book is for

This book is intended for readers who have some experience with R, but does not assume any previous familiarity with Bioconductor. The examples and workflows include all R code required to download data, set up data objects, perform analyses, and visualize results.

For readers who are new to R and Bioconductor, additional useful resources include:

- The [Orchestrating Single-Cell Analysis with Bioconductor (OSCA)](https://osca.bioconductor.org/) book @Amezquita2020, which contains additional details on the analysis steps described here in the context of single-cell RNA sequencing instead of ST data, as well as additional introductory chapters on R and Bioconductor

- The [R for Data Science](https://r4ds.had.co.nz/) online textbook provides an excellent introduction to R

- [Data Carpentry](https://datacarpentry.org/) and [Software Carpentry](https://software-carpentry.org/) provide both introductory and more advanced online lesson materials on R programming, as well as the Unix shell and Git

- The R/Bioconductor Data Science Team at LIBD has a [detailed guide](https://lcolladotor.github.io/bioc_team_ds/rbioconductor-data-science-bootcamps.html#.YQyzLtNKgdk) of free resources and videos you can use to learn more about R and Bioconductor. You might also be interested in some of their [YouTube videos](https://www.youtube.com/c/LeonardoColladoTorres/playlists) and [LIBD rstats club sessions](https://docs.google.com/spreadsheets/d/1is8dZSd0FZ9Qi1Zvq1uRhm-P1McnJRd_zxdAfCRoMfA/edit?usp=sharing), including some on the basics of Bioconductor and the infrastructure for storing gene expression data.


## Bioconductor

[Bioconductor](http://bioconductor.org/) is an **open source** and **open development** project, providing a cohesive and flexible framework for analyzing high-throughput genomics data in R @Huber2015. The Bioconductor project consists of around 2000 contributed R packages, as well as core infrastructure maintained by the Bioconductor Core Team, providing a rich analysis environment for users.

One of the main advantages of Bioconductor is the modularity and open development philosophy. R packages implementing analysis methods are contributed by numerous research groups, with the Bioconductor Core Team coordinating the overall project and maintaining infrastructure such as data class structures and package testing infrastructure. A key feature is that contributed packages use consistent data structures, such as [SingleCellExperiment](http://bioconductor.org/packages/release/bioc/vignettes/SingleCellExperiment/inst/doc/intro.html) or [SpatialExperiment](https://bioconductor.org/packages/SpatialExperiment), which allows packages to interact nicely (e.g. output from one package can easily be provided as input to another package). In addition, contributed Bioconductor packages are expected to have comprehensive documentation, including vignettes (long-form examples including R code).

This modular and open-development approach allows end users to readily combine and substitute analysis methods contributed by different research groups, instead of being "locked in" to using methods or packages maintained by a single group. In a fast-moving field such as single-cell or ST analysis, having the ability to update analysis pipelines to use the latest methods is a crucial advantage. Any research group can contribute new packages implementing new methods to Bioconductor by following the contribution guidelines.


## Contributions

### Contribution guidelines

We welcome suggestions for contributions to OSTA. These should follow the following contribution guidelines, which are intended to maximize the usefulness of OSTA as a streamlined resource for new and advanced readers:

- *Openness*: all methods and tools described should be freely available
- *Neutrality*: workflows should largely rely on standard methods that have been demonstrated to perform well (for example in published independent benchmark papers), to avoid perceived bias towards methods developed by specific groups
- *Bioconductor-based*: workflows should be based on Bioconductor packages, where possible
- *Data structures*: examples should make use of Bioconductor data structures such as [SpatialExperiment](https://bioconductor.org/packages/SpatialExperiment)
- *Testability*: code examples will be tested regularly on the Bioconductor build system, which requires that R package dependencies are available from Bioconductor or CRAN
- *Maintenance*: ongoing commitment to maintain contributed code as part of the regular [Bioconductor release schedule](http://bioconductor.org/developers/)


### Contributions to workflows

In particular, we welcome contributions to [Workflows]. These are chapters in OSTA that each describe a complete workflow, consisting of several individual analysis steps. For example, these may be focused on a specific type of technological platform, or designed to showcase a certain type of analysis.

If you want to demonstrate a new method (e.g. a method for spatially aware clustering implemented in a Bioconductor package you have developed), we recommend adding it to the relevant workflow(s) to compare against other methods performing a similar task. If you decide to add to an existing workflow, the new method should run relatively fast (seconds to minutes) on the given dataset, due to limitations on the time it takes to build the book.

If you decide to contribute to an existing or new workflow, we ask that you first open a [GitHub issue](https://github.com/lmweber/OSTA-book/issues) to propose the idea along with an outline of your proposed examples or workflow. The OSTA team can then give you initial feedback and you can proceed to make a pull request on GitHub. Where possible, we recommend re-using code from the individual analysis chapters within OSTA (e.g. for standard analysis steps), which will make the examples easier for readers to follow.

