# Space Ranger (Visium)


## Overview
Space Ranger is a set of analysis pipelines for processing Visium sequence data (fasta files) with high resolution microscope images of tissue. In short, it maps the transcriptomic reads to the microscope image of the tissue from which the reads were obtained. Space Ranger includes 5 piplines but we only use one for out work with Visium and this the "spaceranger count" pipeline. 
You can find more information about the various pipelines [here](https://support.10xgenomics.com/spatial-gene-expression/software/pipelines/latest/what-is-space-ranger).


## Installation

Installing Space is fairly simple and can be completed about 3 steps which can be found [here](https://support.10xgenomics.com/spatial-gene-expression/software/pipelines/latest/installation).


## Memory issues


## Run spaceranger count

This pipeline takes the microscope image and fastq files and performs sequence aligment, tissue detection and alignment using the fiducial frame, and barcode/UMI counting. The most important output is the gene-spot matrix which we combine with other data generated by Vistoseq (see below) into a Spatial Experiment object for downstream analysis. 

A more detailed description of this pipeline can be found [here](https://support.10xgenomics.com/spatial-gene-expression/software/pipelines/latest/using/count). 

Here is an example of the parameters used to run spaceranger count. 

```{r, echo=FALSE, out.width = "40%", fig.align="center", fig.cap="Figure caption."}
knitr::include_graphics("images/space_ranger_input.png")
```

## Output files

The Space Ranger pipeline outputs the following files. The ones we use for downstream analysis are the files contained the the raw_features_bc_matrix folder.  

```{r, echo=FALSE, out.width = "40%", fig.align="center", fig.cap="Figure caption."}
knitr::include_graphics("images/space_ranger_output.png")
```


## Web summary .html file

The web summary is the first file we look at after Space Ranger is done running because it tells us if the run was successful. It provides general quality control statistics and visualizations.  
```{r, echo=FALSE, out.width = "75%", fig.align="center", fig.cap="Figure caption."}
knitr::include_graphics("images/web_summary.png")
```
The cloupe file is also useful for checking the quality of the data. There's more information on this file the Loupe Browswer section of this book. 

## Import outputs into R

We use the [Spatial Experiment](https://bioconductor.org/packages/devel/bioc/html/SpatialExperiment.html) package to store the data from the Space Ranger and Vistoseq (see below) in one convenient spatial experiment object (spe). Below is an example of the code we use to build the spe object.

```{r, echo=FALSE, out.width = "75%", fig.align="center", fig.cap="Figure caption."}
knitr::include_graphics("images/build_spe.png")
```
## Vistoseq for quantifying cells per spot
Vistoseg is a MATLAB pipeline used to process, analyze, and interactively visualize the high-resolution histology images from the Visium experiment form 10x genomics. The output from VistoSeg is used for downstream analyses. The main steps in VistoSeg are as follows:
Split Image: This step is used to split the large Visium histology whole side image into individual capture areas, creating 4 individual .tif files.
Nuclei Segmentation: This is a two-step process where nuclei segmentation is performed for images, nuclei segmentation is followed by a refining step to get cleaner image of the nuclei segmentation.
Spaceranger
CountNuclei: This steps is performed to get the count of cells/nuclei per Visium spot for each capture area. The output is saved in a excel (tissue_spot_counts.csv) file.
The data for nuclei count for the tissue per visium spot is incorporated with the outputs of spaceranger by being stored in a spatial experiment object or used for downstream analysis.

For more information about Vistoseq please see [Vistoseg](http://research.libd.org/VistoSeg/index.html)
