# Space Ranger (Visium)


## Overview

Space Ranger is a set of analysis pipelines for processing 10x Genomics Visium sequence data (`fastq` files) with high resolution microscope images of tissue. In short, it maps the transcriptomic reads to the microscope image of the tissue from which the reads were obtained. Space Ranger includes 5 pipelines but we only use one for our work with Visium -- this is the `spaceranger count` pipeline. 

You can find more information about the various pipelines [here](https://support.10xgenomics.com/spatial-gene-expression/software/pipelines/latest/what-is-space-ranger).


## Installation

Installing Space Ranger is fairly simple and can be completed in around 3 steps which can be found [here](https://support.10xgenomics.com/spatial-gene-expression/software/pipelines/latest/installation).


## Run spaceranger count

This pipeline takes the microscope image and `fastq` files and performs sequence alignment, tissue detection and alignment using the fiducial frame, and barcode/UMI counting. The most important output is the gene-spot matrix which we combine with other data generated by VistoSeg (see below) into a `SpatialExperiment` object for downstream analysis. 

A more detailed description of this pipeline can be found [here](https://support.10xgenomics.com/spatial-gene-expression/software/pipelines/latest/using/count). 

Here is an example of the parameters used to run `spaceranger count`. 

```{r, echo=FALSE, out.width = "40%", fig.align="center", fig.cap="Figure caption."}
knitr::include_graphics("images/space_ranger_input.png")
```


## Output files

The Space Ranger pipeline outputs the following files. The ones we use for downstream analysis are the files contained in the `raw_features_bc_matrix` folder.  

```{r, echo=FALSE, out.width = "40%", fig.align="center", fig.cap="Figure caption."}
knitr::include_graphics("images/space_ranger_output.png")
```


## Web summary .html file

The web summary `.html` is the first file we look at after Space Ranger is done running because it tells us if the run was successful. It provides general quality control statistics and visualizations.  

```{r, echo=FALSE, out.width = "75%", fig.align="center", fig.cap="Figure caption."}
knitr::include_graphics("images/web_summary.png")
```

The `.cloupe` file is also useful for checking the quality of the data. There is more information on this file the [Loupe Browser (Visium)] section of this book. 


## Import outputs into R

We use the [SpatialExperiment](https://bioconductor.org/packages/SpatialExperiment) package to store the data from Space Ranger and VistoSeg (see below) in one convenient object (`spe`). Below is an example of the code we use to build the `spe` object.

```{r, echo=FALSE, out.width = "75%", fig.align="center", fig.cap="Figure caption."}
# knitr::include_graphics("images/build_spe.png")
```


## VistoSeg for quantifying cells per spot

VistoSeg is a MATLAB pipeline used to process, analyze, and interactively visualize the high-resolution histology images from the 10x Genomics Visium experiment. The output from VistoSeg is used for downstream analyses. The main steps in VistoSeg are as follows:

- Split Image: This step is used to split the large Visium histology whole slide image into individual capture areas, creating 4 individual `.tif` files.
- Nuclei Segmentation: This is a two-step process where nuclei segmentation is performed for images. Nuclei segmentation is followed by a refining step to get a cleaner image of the nuclei segmentation.
- Space Ranger
- CountNuclei: This steps is performed to get the count of cells/nuclei per Visium spot for each capture area. The output is saved in a spreadsheet file (`tissue_spot_counts.csv`).

The data for nuclei counts for the tissue per Visium spot is incorporated with the outputs of Space Ranger by being stored in the `SpatialExperiment` object or used for downstream analysis.

For more information about VistoSeg please see [VistoSeg](http://research.libd.org/VistoSeg/index.html)

